Octokit.Reactive/Helpers/ConnectionExtensions.cs

```diff
@@ -37,6 +37,15 @@ public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connectio
             return GetPages(url, parameters, (pageUrl, pageParams) => connection.Get<List<T>>(pageUrl, pageParams, accepts).ToObservable());
         }
 
+        public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connection, Uri url, IDictionary<string, string> parameters, string accepts, ApiOptions options)
+        {
+            return GetPagesWithOptions(url, parameters, options, (pageUrl, pageParams, o) =>
```

Comment:
This method isn't paginating as I'd hoped - it's continuing to make the same request forever. Looking into what I've missed here. Here's the test I'm using, feel free to pull it into this PR:

```
[IntegrationTest]
public async Task CanRetrievePagesOfPublicStarWatchers()
{
    var owner = "octokit";
    var name = "octokit.net";

    var firstPageOptions = new ApiOptions
    {
        PageSize = 5,
        StartPage = 1
    };

    var firstPage = await _fixture.GetAllStargazersWithTimestamps(owner, name, firstPageOptions);

    var secondPageOptions = new ApiOptions
    {
        PageSize = 5,
        StartPage = 2
    };

    var secondPage = await _fixture.GetAllStargazersWithTimestamps(owner, name, secondPageOptions);
    Assert.Equal(5, firstPage.Count);
    Assert.Equal(5, secondPage.Count);

    Assert.NotEqual(firstPage[0].User.Id, secondPage[0].User.Id);
    Assert.NotEqual(firstPage[1].User.Id, secondPage[1].User.Id);
    Assert.NotEqual(firstPage[2].User.Id, secondPage[2].User.Id);
    Assert.NotEqual(firstPage[3].User.Id, secondPage[3].User.Id);
    Assert.NotEqual(firstPage[4].User.Id, secondPage[4].User.Id);
}
```